gstreamer/src/query.rs

```diff
@@ -219,73 +327,123 @@ impl ToOwned for QueryRef {
     }
 }
 
-pub enum QueryView<T> {
-    Position(Position<T>),
-    Duration(Duration<T>),
-    Latency(Latency<T>),
-    Jitter(Jitter<T>),
-    Rate(Rate<T>),
-    Seeking(Seeking<T>),
-    Segment(Segment<T>),
-    Convert(Convert<T>),
-    Formats(Formats<T>),
-    Buffering(Buffering<T>),
-    Custom(Custom<T>),
-    Uri(Uri<T>),
-    Allocation(Allocation<T>),
-    Scheduling(Scheduling<T>),
-    AcceptCaps(AcceptCaps<T>),
-    Caps(Caps<T>),
-    Drain(Drain<T>),
-    Context(Context<T>),
-    Other(Other<T>),
+#[derive(Debug)]
+pub enum QueryView {
+    Position(Position),
+    Duration(Duration),
+    Latency(Latency),
+    Jitter(Jitter),
+    Rate(Rate),
+    Seeking(Seeking),
+    Segment(Segment),
+    Convert(Convert),
+    Formats(Formats),
+    Buffering(Buffering),
+    Custom(Custom),
+    Uri(Uri),
+    Allocation(Allocation),
+    Scheduling(Scheduling),
+    AcceptCaps(AcceptCaps),
+    Caps(Caps),
+    Drain(Drain),
+    Context(Context),
+    Other(Other),
     __NonExhaustive,
 }
 
+impl From<Query> for QueryView {
```

Comment:
With this you can take a `Query` with refcount > 1, get a `QueryView`, then call `&mut QueryView` functions. And those would fail. Or am I missing something?

Reply:
I think you're right. I'll look into that tomorrow.

Reply:
Thanks :) Generally this all looks very good usage-wise, but we need to be careful to not break things and to allow code that should not be able to compile